api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClientImpl.java

```diff
@@ -169,7 +171,12 @@ public void deleteCluster(String googleProject, String clusterName) {
   @Override
   public Cluster getCluster(String googleProject, String clusterName) {
     ClusterApi clusterApi = clusterApiProvider.get();
-    return retryHandler.run((context) -> clusterApi.getCluster(googleProject, clusterName));
+    try {
+      return retryHandler.runAndThrowChecked(
+          (context) -> clusterApi.getCluster(googleProject, clusterName));
```

Comment:
nit: maybe rename lambda variable to `unusedContext`. You can drop the parenthesis for single-arg lambdas FWIW.